sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFEvaluators.scala

```diff
@@ -129,7 +129,7 @@ class HiveGenericUDFEvaluator(
   override def returnType: DataType = inspectorToDataType(returnInspector)
 
   def setArg(index: Int, arg: Any): Unit =
```

Comment:
I think we should make it lazy at the very beginning, `arg` here is already materialized.